components/collector/src/source_collectors/gitlab/pipeline_duration.py

```diff
@@ -0,0 +1,22 @@
+"""GitLab CI-pipeline duration collector."""
+
+from collector_utilities.type import URL, Value
+from model import SourceResponses
+
+from .base import GitLabPipelineBase
+
+
+class GitLabPipelineDuration(GitLabPipelineBase):
+    """GitLab CI-pipeline duration collector."""
+
+    async def _get_source_responses(self, *urls: URL) -> SourceResponses:
+        """Override to get the pipeline details."""
+        responses = await super()._get_source_responses(*urls)
+        pipelines = await self._pipelines(responses)
+        api_url = await self._gitlab_api_url(f"pipelines/{pipelines[0].id}")
```

Comment:
Doesn't the high default `lookback_days` mean that this will be a very slow KPI? Should we protect users against configuring something dumb?

Reply:
Yeah, good thinking. Unfortunately the GitLab `lookback_days` parameter is also used for other metrics (`change_failure_rate`, `failed_jobs`, `job_runs_within_time_period`, `source_up_to_dateness`, `unused_jobs`) where it does make sense to have a higher default value.

The Azure DevOps collector does have two separate loopback parameters, one for pipelines and one for issues. We should make the same distinction for GitLab. I'll add an tech issue to the backlog.

Reply:
Strike that. I'll change it right away. As he pipeline duration Is a new metric I can just as well introduce a new parameter now.